roles/openstack/neutron/tasks/cleanup.yml

```diff
@@ -0,0 +1,30 @@
+---
+# This file is used to cleanup OpenStack neutron resources
+
+- name: Cleanup routers
+  vars:
+      prefix: "{{ provisioner.prefix | default(uuid) }}-"
+  os_router:
+      interfaces: "{{ item.value.attach_subnets | map('add_prefix', prefix) | join(',') }}"
+      name: "{{ prefix }}-{{ item.value.name | default(router_name) }}"
```

Comment:
prefix already have "-" at the end.
How about: `{{ [prefix, item.value.name] | join("-") }}` ?